src/System.Text.Json/src/System/Text/Json/CustomUncheckedBitArray.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json
+{
+    // Inspired by BitArray
+    // https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/BitArray.cs
+    internal struct CustomUncheckedBitArray
```

Comment:
What does this have to do with ReadOnlySequence?

Reply:
It doesn't. I just happen to add ReadOnlySequeunce on top of the changes from this PR: https://github.com/dotnet/corefx/pull/33378